### PR TITLE
Update pki docs

### PR DESCRIPTION
**Pull request by qrkourier**

### CODE REVIEW

File: docusaurus/docs/learn/core-concepts/security/authentication/50-external-id-claims.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-# External JWT Signers
+# External ID Claims
```
Comment (andrewpmartinez) [inline docusaurus/docs/learn/core-concepts/security/authentication/50-external-id-claims.md:1]: I wouldn't suggest introducing a new term, "External ID Claim" without defining what it is. That term doesn't appear anywhere I know of other than here (i.e. not in APIs, CLIs, or code).

It seems like you mean to cover both External JWT Signers and X509 Claims here. If you are going to do that I suggest a list at the top explaining the new term and the two things described below so that the reader has a framework to start out with.

File: docusaurus/docs/guides/alt-server-certs.md [hunk 2]
```diff
@@ -4,68 +4,75 @@ sidebar_label: Public CA Certs
...
+Alternative server certificates are most relevant to Ziti networks that use [BrowZer](https://blog.openziti.io/series/browzer). BrowZer is an implementation of Ziti's external ID claim feature. Unlike enrollment, which configures an explicit root of trust for mTLS, an external ID claim allows short-lived authentication instruments, e.g., JWT or x509 client cert, issued by a trusted identity provider to "claim" an existing Ziti identity. This is how a BrowZer user is able to connect to a Ziti service with an authorization token from an OIDC provider.
```
Comment (andrewpmartinez) [inline docusaurus/docs/guides/alt-server-certs.md:9]: `external id claim` is said here, but that term has no standard meaning and it doesn't link to anywhere to explain what it is.

Comment (andrewpmartinez) [inline docusaurus/docs/guides/alt-server-certs.md]: `external id claim` should probably link to someplace that describes it clearly rather than trying to describe it quickly w/ the `e.g.` portion.

File: docusaurus/docs/guides/alt-server-certs.md [hunk 3]
```diff
@@ -4,68 +4,75 @@ sidebar_label: Public CA Certs
...
-      cert: /etc/pki/ca/example.com/certs/client1.pem  # required but unused
+identity:
+  cert: ctrl-plane-client.crt                  # user cert from ctrl plane CA
+  server_cert: ctrl-plane-server.crt           # server cert from ctrl plane CA with SAN "ctrl.ziti.example.com"
+  key: ctrl-plane-client.key                   # private key of client cert
+  server_key: ctrl-plane-server.key            # private key of server certs
+  ca: ctrl-plane-trust-bundle.pem              # bundle of root CAs that may be used to verify this identity's leaf cert chains
+  alt_server_certs:
+    - server_cert: lets_encrypt.cert.pem       # server cert chain from Let's Encrypt with SAN "client-pub.ziti.example.com"
+    - server_key: lets_encrypt.key.pem         # private key of alt server cert
```
Comment (andrewpmartinez) [inline docusaurus/docs/guides/alt-server-certs.md]: This snippet is incorrect. This is creating two invalid `alt_server_certs` one with just a cert and another with just a key. Each line prefixed w/ a dash starts a new object.

It should be:

```
  alt_server_certs:
    - server_cert: lets_encrypt.cert.pem       # server cert chain from Let's Encrypt with SAN "client-pub.ziti.example.com"
      server_key: lets_encrypt.key.pem
```

File: docusaurus/docs/guides/alt-server-certs.md [hunk 4]
```diff
@@ -4,68 +4,75 @@ sidebar_label: Public CA Certs
...
-  ca: /etc/pki/ca/example.com/certs/ca.pem
+  cert: ctrl-plane-client.crt                  # filepath to write client cert from controller's edge enrollment signer during enrollment
+  server_cert: ctrl-plane-server.crt           # filepath to write server cert from edge enrollment signer during enrollment
+  key: ctrl-plane.key                          # filepath to write generated private key for leaf certs during enrollment
+  ca: ctrl-plane-trust-bundle.pem              # filepath to write known CA certs during enrollment
   alt_server_certs:
-    - server_cert: /etc/letsencrypt/live/example.com/fullchain.pem
-      server_key: /etc/letsencrypt/live/example.com/privkey.pem
+    - server_cert: lets_encrypt.cert.pem       # server cert from Let's Encrypt with a distinct DNS SAN
+    - server_key: lets_encrypt.key.pem         # private key of alt server cert
```
Comment (andrewpmartinez) [inline docusaurus/docs/guides/alt-server-certs.md]: Another extra dash creating two invalid `alt_server_cert`s

File: docusaurus/docs/learn/core-concepts/pki.md [hunk 5]
```diff
@@ -1,89 +1,56 @@
...
-All Ziti Networks leverage [Public Key Infrastructure (PKI)](https://en.wikipedia.org/wiki/Public_key_infrastructure) to
-provide secure network connections.  This page is not intended to be a comprehensive guide. What it is, is a set of
-rules that must be followed to properly configure a Ziti Network. If there are issues when connecting any portion
-of a Ziti Network to another - this page should serve as a starting point of understanding.
+[Public Key Infrastructure](https://en.wikipedia.org/wiki/Public_key_infrastructure) (PKI) is used to establish trust between all Ziti components. The Ziti controller enrolls identities with a PKI that it manages, and externally-managed PKIs may be verified with the Ziti controller and used for enrollment. Enrollment is best suited for long-lived client certificate fingerprint authentication.
 
-:::note
-This article is about managing your own PKI. There's a guide for using public CA certificates.
-:::
+Alternatively to enrollment, external PKIs may be trusted to sign authentication instruments that are used to [claim an existing Ziti identity](./security/authentication/50-external-id-claims.md).
```
Comment (andrewpmartinez) [inline docusaurus/docs/learn/core-concepts/pki.md]: What is an "authentication instrument"? If you mean a cryptographically signed document like a jWT/x509.Certifciate, say so plainly.

Readers may come to any specific page not knowing about any other part of the documentation and may not understand abstract or indirect language.

File: docusaurus/docs/learn/core-concepts/pki.md [hunk 6]
```diff
@@ -1,89 +1,56 @@
...
-The Ziti Network allows the operator to declare any trust anchors as valid. This means Ziti does not need to be
-configured with a full chain of certificates which link fully back to a root CA. A configuration using a full chain back
-to a root CA is of course supported but it is not explicitly required.  This allows the operator to configure a Ziti
-Network using one or more chains of trust back to the provided trust anchors.  The sections below will describe where
-these trust anchors can be configured.
+### Ziti Controller
 
-Ziti Network components are required to present a certificate to other Ziti Network components during the connection
-establishment. This certificate will need to be valid per the configured trust anchor store being connected to.
+The certificate authority (CA) in a Ziti network's PKI is the controller's edge enrollment CA. The edge enrollment CA issues client authentication certificates to identities during identity enrollment. The edge enrollment CA issues client and server authentication certificates to routers during router enrollment.
```
Comment (andrewpmartinez) [inline docusaurus/docs/learn/core-concepts/pki.md]: The first sentence, "The CA in a Ziti's network's PKI is the controller's edge enrollment CA" is superfluous and also technically incorrect. In any PKI there can be more than 1 CA.

I suggest rewording this to state that the Ziti Controller is configured with a CA that is used as the enrollment CA.

File: docusaurus/docs/learn/core-concepts/pki.md [hunk 7]
```diff
@@ -1,89 +1,56 @@
...
+The controller provides these TLS servers:
 
-**ca**: A file representing a group of certificates with one or more certificate chains terminating at a trust anchor.
-When a Ziti Network component connects to the Ziti Controller and offers a certificate for validation the incoming
-connection is checked to see if it signed by a trust anchor specified in this file.
+* The control plane API is consumed by routers and presents the certificate defined in `identity.server_cert`.
+* One or more web listeners provide bindings for the controller's REST APIs and other web services. Each web listener presents the certificate defined in its own `web[].identity.server_cert` property, or defaults to the root-level `identity.server_cert`.
+
+:::note
+The private key of the web listener to which the client API is bound is used to sign edge enrollment tokens. During enrollment, the identity or router in possession of a token will verify that it was signed by the same key that is backing the server certificate of the controller's client API. This allows the enrolling identity to confirm the client API URL is correct before requesting authentication certificate(s).
```
Comment (andrewpmartinez) [inline docusaurus/docs/learn/core-concepts/pki.md]: Should probably use the full name of the API in all instanced: Edge Client API and then link to its documentation.

File: docusaurus/docs/learn/core-concepts/pki.md [hunk 8]
```diff
@@ -1,89 +1,56 @@
...
+::: 
 
-**key**: Also referred to as the [private key](https://en.wikipedia.org/wiki/Symmetric-key_algorithm). It is generated
-first and used to produce the certificates specified in the `cert` and `server_cert` fields of the Ziti Controller
-configuration file.
+The trust bundle defined in controller config property `identity.ca` is used by the controller to bundle known CAs' certificates. This bundle is downloaded by clients during enrollment from the client API in `/edge/client/v1/.well-known/est/cacerts` and subsequently used to verify controller and router server certificate chains. These known CAs are never used by the controller to verify client certificates which are always from the controller's own edge enrollment CA or a known, verified external CA.
```
Comment (andrewpmartinez) [inline docusaurus/docs/learn/core-concepts/pki.md]: `external CAs` are called `3rd Party CAs` in the code, API. The CLI calls them simply `ca` (which is unhelpful).

In the name of having a universal domain language, I suggest using 3rd Party CA here.


`...or a known, verified external CA represented as a [3rd Party Ca](link to 3rd Party CAs)`

File: docusaurus/docs/learn/core-concepts/pki.md [hunk 9]
```diff
@@ -1,89 +1,56 @@
...
-#### PKI Configuration
-
-The `identity` section of the Edge Router configuration is used by the Edge Router when connections are
-established to or from the other components of a Ziti Network. There are four sections in the identity block:
-`cert`, `server_cert`, `key`, `ca`.
-
-**ca**: A file representing a group of certificates with one or more certificate chains terminating at a trust anchor.
-When a Ziti Network component connects to the Edge Router and offers a certificate for validation the incoming
-connection is checked to see if it signed by a trust anchor specified in this file.
+provided exists the Edge Router will use it and the other fields will be **regenerated and overwritten** as necessary, if the filesystem is writable.
```
Comment (andrewpmartinez) [inline docusaurus/docs/learn/core-concepts/pki.md:47]: What happens if the file system isn't writable?

File: docusaurus/docs/learn/core-concepts/security/authentication/10-third-party-cas.md [hunk 10]
```diff
@@ -79,89 +76,55 @@ The [Edge Management API](/docs/reference/developer/api#edge-management-api) acc
...
-and parsing allows for a single value to yield multiple claims.
-
-#### Location
-
-Location configuration sources value(s) from the x509 certificate
 
-- `COMMON_NAME` - the common name of the certificate
-- `SAN_URI` - SAN URI fields
-- `SAN_EMAIL` - SAN email fields
+The enrollment JWT can be retrieved by fetching the created identity by ID and parsing out the `enrollment.ottca.jwt` field.
```
Comment (andrewpmartinez) [inline docusaurus/docs/learn/core-concepts/security/authentication/10-third-party-cas.md]: This is planned for removal. The way to get it will be to request `GET /identities/:id/enrollments`.  Which lists all outstanding enrollments for a specific identity.

File: docusaurus/docs/learn/core-concepts/security/authentication/50-external-id-claims.md [hunk 11]
```diff
@@ -1,4 +1,6 @@
...
+
```
Comment (andrewpmartinez) [inline docusaurus/docs/learn/core-concepts/security/authentication/50-external-id-claims.md:2]: It seems weird that there is a 3rd Party CA page and not an External JWT SIgner page. This page was the latter, but it was merged with a meta-concern of mapping identities to external IdPs.

It obscures External JWT SIgners and merges two separate implementations into a meta concern that should be its own page. I think there is space in the documentation for clinically describing what things are and how they work (reference documentation) and another place for describing meta-concerns and how to get things done (admin/user documentation).

I feel that we have lost the reference documentation aspect in the security section with these changes and are now attempting to make it admin/user documentation. I would much rather this page be largely reverted, and a new page or section be created for this new page's goals.

The good thing about reference documentation is that it provides quick answers to those who want to know what an argument or process does. It also allows admin/user documentation to reference it, so it doesn't have to spell out all the details.

The more I think about it, the more I think this perspective explains why our documentation seems so disjointed. Some authors are describing how things work/their inputs/outputs (largely the engineering team), and others are trying to describe how to do specific common tasks (everyone else). The documentation gets put in the same places and works poorly.

File: docusaurus/docs/learn/core-concepts/security/authentication/50-external-id-claims.md (hunk 1 above)
Comment (qrkourier) [reply docusaurus/docs/learn/core-concepts/security/authentication/50-external-id-claims.md:1]: Good callout. That's what I meant. It makes more sense to document the types of claims in this article, and types of enrollments in the enrollments article, because claims and enrollments are fundamentally different ways of authenticating an Edge Identity. A brief, user-centric comparison might be helpful too, e.g., "Enrollments are typically used for long-lived credentials, and claims are well-suited for short-lived credentials."

File: docusaurus/docs/guides/alt-server-certs.md (hunk 2 above)
Comment (qrkourier) [reply docusaurus/docs/guides/alt-server-certs.md]: https://github.com/openziti/ziti-doc/pull/592/commits/fa2b8398e4bc4fcf042dd60a580bfd60e7f2de49

File: docusaurus/docs/guides/alt-server-certs.md (hunk 3 above)
Comment (qrkourier) [reply docusaurus/docs/guides/alt-server-certs.md]: https://github.com/openziti/ziti-doc/pull/592/commits/ae9a3c7925e41ec6e6e220432fac91fc132b6254

File: docusaurus/docs/guides/alt-server-certs.md (hunk 4 above)
Comment (qrkourier) [reply docusaurus/docs/guides/alt-server-certs.md]: https://github.com/openziti/ziti-doc/pull/592/commits/ae9a3c7925e41ec6e6e220432fac91fc132b6254

File: docusaurus/docs/learn/core-concepts/pki.md (hunk 5 above)
Comment (qrkourier) [reply docusaurus/docs/learn/core-concepts/pki.md]: https://github.com/openziti/ziti-doc/pull/592/commits/7a604fb8b5d2a44a30c52b568d3480e9f0c99d73

File: docusaurus/docs/learn/core-concepts/pki.md (hunk 6 above)
Comment (qrkourier) [reply docusaurus/docs/learn/core-concepts/pki.md]: https://github.com/openziti/ziti-doc/pull/592/commits/9fb0ef9913d8b2fd566a287e1f279c734297269c

File: docusaurus/docs/learn/core-concepts/pki.md (hunk 7 above)
Comment (qrkourier) [reply docusaurus/docs/learn/core-concepts/pki.md]: https://github.com/openziti/ziti-doc/pull/592/commits/91d6c14e6a1befcce4631a44960d58590e1ba034

File: docusaurus/docs/learn/core-concepts/pki.md (hunk 9 above)
Comment (qrkourier) [reply docusaurus/docs/learn/core-concepts/pki.md:47]: Nothing happens if it's not writable. Should we say that?

File: docusaurus/docs/learn/core-concepts/pki.md (hunk 8 above)
Comment (qrkourier) [reply docusaurus/docs/learn/core-concepts/pki.md]: https://github.com/openziti/ziti-doc/pull/592/commits/a693302bc4cb86c09550fccf1bef2cbe156f234c

File: docusaurus/docs/learn/core-concepts/security/authentication/10-third-party-cas.md (hunk 10 above)
Comment (qrkourier) [reply docusaurus/docs/learn/core-concepts/security/authentication/10-third-party-cas.md]: That explains why I was struggling with `{{baseUrl}}/enrollments?filter=identityId%3D%22LcTkRcyVdM%22` not filtering enrollments by identity `identityId`.

Comment (qrkourier) [reply docusaurus/docs/learn/core-concepts/security/authentication/10-third-party-cas.md]: https://github.com/openziti/ziti-doc/pull/592/commits/7d2febee2161190ed08209c5c7d9750219aa9f91

File: docusaurus/docs/learn/core-concepts/security/authentication/50-external-id-claims.md (hunk 11 above)
Comment (qrkourier) [reply docusaurus/docs/learn/core-concepts/security/authentication/50-external-id-claims.md:2]: I agree we should keep nudging docs toward an organization by audience and intent, e.g., learning (tutorials and explanations) vs. using (references and guides). We've already begun to structure the docs in those ways, but there's still more that we can do. The most helpful model we've stumbled upon so far is this one: https://documentation.divio.com/. How does that model land with you? 

Specifically, we could, as you suggest, organize the reference content separately from concept-level, explanatory articles.

File: docusaurus/docs/guides/alt-server-certs.md
Comment (qrkourier) [inline docusaurus/docs/guides/alt-server-certs.md:1]: I've moved the changes to this file into a separate pull request: https://github.com/openziti/ziti-doc/pull/604

**Comment by qrkourier** [comment]
This branch partially answers the need for documenting external id claims described in issues https://github.com/openziti/ziti-doc/issues/411 and https://github.com/openziti/ziti-doc/issues/177

**Comment by qrkourier** [comment]
Thank you for the thoughtful review. I'll close this in favor of a new attempt to document what the administrator needs to know to use external ID claims successfully.
